opacus/tests/grad_samples/dp_rnn_test.py

```diff
@@ -76,8 +76,5 @@ def test_rnn(
         if using_packed_sequences:
             x = _gen_packed_data(N, T, D, batch_first, packed_sequences_sorted)
         else:
-            if batch_first:
-                x = torch.randn([N, T, D])
-            else:
-                x = torch.randn([T, N, D])
+            x = torch.randn([N, T, D])
```

Comment:
looks like the if-else condition is not useful, as we already swap the input in https://github.com/pytorch/opacus/blob/3f3cc88340b2476cf685a3c2eb1f03b70ccfff9b/opacus/tests/grad_samples/common.py#L122

Reply:
That's incorrect.

Input data should always match `batch_first` flag.

a) Line you quote only works in `compute_microbatch_grad_sample` branch - that's half of the test (reminder: the test runs `compute_microbatch_grad_sample` and `compute_opacus_grad_sample` to make sure they produce identical results.)

b) If you look further down to lines 128-130 https://github.com/pytorch/opacus/blob/3f3cc88340b2476cf685a3c2eb1f03b70ccfff9b/opacus/tests/grad_samples/common.py#L128-L130

we then put batch dimension back, while respecting `batch_first` flag

Reply:
Hmm good point. Do you have any idea why the tests pass then? Is it because the input is just randomly generated?

Reply:
I still don't understand why you think this change is not correct. We are changing the `x` on line 122 of common.py, since you wrote

> Input data should always match batch_first flag.